hy/core/shadow.hy

```diff
@@ -138,11 +138,17 @@
 (defn not-in [x y]
   (not-in x y))
 
+(defn get [thing key1 &rest keys]
+  (for* [k (+ (, key1) keys)]
+    (setv thing (get thing k)))
+  thing)
```

Comment:
> version using reduce

I don't think `reduce` would be faster considering the overhead for Python's function calls.

> But why use the if?

So we immediately stop processing if the keys are empty, which should speed up the common 1-key case. I'm actually not certain if that helps. Python might do this check anyway in the for loop. I can profile and maybe disassemble the bytecode to see if it matters.